### PR TITLE
TCG Card search window

### PR DESCRIPTION
**Pull request by Blackshadow8910**
This pr will add the window for searching for cards with basic functionality, eg searching by name.

### CODE REVIEW

**Comment by Blackshadow8910** [comment]
As a note for future implementation: I think the separate parts of the card search UI should be individual views, so we can modularize it to reuse the CardSearchView in the other features as well. As in, the info panel on the right should be its own class, and be can instantiate it for the card search tab and connect it to the cardsearchview via listeners or something, and then add another cardsearchview to the deck builder tab where we can then add a different side panel that instead allows adding to the deck,

**Comment by Blackshadow8910** [comment]
Alright, i took the cardview class out and isolated it, so we can use that in any place we want to display a list of cards. So, if anyone wants to work on the deck builder or anything, use that.

The ui for card searching still needs to be revamped, I intend to change the layout from gridbag to borderlayout with hard coded sizes to prevent the right panel from shifting when selecting cards, or somebody else can implement it if they have time. The search bar i also want to move to its own class.

File: src/main/java/app/Main.java [hunk 1]
```diff
@@ -26,23 +33,23 @@ public static void main(String[] args) {
...
         // Other
 
         GUIManager guiManager = new GUIManager();
 
+        CardSearchViewModel cardSearchViewModel = new CardSearchViewModel();
         LoginViewModel loginViewModel = new LoginViewModel();
         AppViewModel appViewModel = new AppViewModel();
+        SignupViewModel signupViewModel = new SignupViewModel();
+
+        CardSearchView cardSearchView = CardSearchUseCaseFactory.create(cardSearchViewModel, pokemonCardDAO, imageDAO);
```
Comment (yujusti) [inline src/main/java/app/Main.java:45]: I see you've added a CarSearchView Model, is this within the App interface?

File: src/main/java/app/usecase_factories/AppUseCaseFactory.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import app.GUIManager;
+
+import java.io.IOException;
+
+public class AppUseCaseFactory {
+
+    /** Prevent instantiation. */
+    private AppUseCaseFactory() {}
+
+    public static AppView create(AppViewModel appViewModel, CardSearchView cardSearchView) {
```
Comment (yujusti) [inline src/main/java/app/usecase_factories/AppUseCaseFactory.java:23]: I can add more to this for the App View User Interface

File: src/main/java/data_access/pokemon/TestCardDataAccessObject.java [hunk 3]
```diff
@@ -36,6 +36,21 @@ public TestCardDataAccessObject() {
             "pikamon",
             "pica",
             List.of("electric")));
+        cards.add(new PokemonCard(
+            "Goober", 
+            "gober",
+            "stu",
+            List.of("what")));
+        cards.add(new PokemonCard(
+            "Frep",
```
Comment (yujusti) [inline src/main/java/data_access/pokemon/TestCardDataAccessObject.java:45]: Is this a test of adding pokemons into the list?

File: src/main/java/interface_adapters/signup/SignupPresenter.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    private final LoginViewModel loginViewModel;
+
+    public SignupPresenter(SignupViewModel signupViewModel,
+                           LoginViewModel loginViewModel) {
+        this.signupViewModel = signupViewModel;
+        this.loginViewModel = loginViewModel;
+    }
+
+    @Override
+    public void presentSuccess(SignupOutputData data) {
```
Comment (yujusti) [inline src/main/java/interface_adapters/signup/SignupPresenter.java:19]: Good implementation

**Review by yujusti — APPROVED** [review]
Good Implementation Oliver, I'm looking forward to building on this!

**Pull request merged**